opensrp-eusm/src/main/java/org/smartregister/eusm/viewholder/TaskRegisterViewHolder.java

```diff
@@ -88,6 +89,8 @@ public void setProductImage(@Nullable TaskDetail taskDetail) {
                 this.productImageView.setImageDrawable(ResourcesCompat.getDrawable(context.getResources(), R.drawable.service_point_check_thumbnail, context.getTheme()));
             } else if (AppConstants.NonProductTasks.WAREHOUSE_CHECK.equalsIgnoreCase(taskDetail.getEntityName())) {
                 this.productImageView.setImageDrawable(ResourcesCompat.getDrawable(context.getResources(), R.drawable.service_point_check_thumbnail, context.getTheme()));
+            } else if (AppConstants.TaskCode.FIX_PROBLEM_CONSULT_BENEFICIARIES.equalsIgnoreCase(taskDetail.getTaskCode())) {
```

Comment:
Whats the value of `taskDetail.getEntityName()` here? If its the same as taskCode kindly use the entityName.